### ISSUE TITLE
[Feature Request] slidesToSlide as responsive option with dots per "page"

### ISSUE THREAD

**Issue by tpinne** [YIZHUANG/react-multi-carousel#31]
It would be awesome if the option `slidesToSlide` could be set in the responsive settings. This would easily enable to slide by the visible amount of slides on each breakpoint. In our use case we want always to slide all visible slides at once and show the next "page".

Additionally it would be great if the dots could have a mode to not show the number of slides but the number of "pages" without having to implement the logic with a custom dot component. You see this often in other libraries automatically if their "`slidesToSlide`" option is set to a value > 1.

**Comment by YIZHUANG** [comment]
Hi, the first one is reasonable. Can you elaborate more on the second one with the dots?

**Comment by tpinne** [comment]
Currently if you have for example 9 slides in the carousel there will be 9 dots shown. Regardless what the config for `items` or `slidesToSlide` are set to for the specific breakpoint.

Lets say you set the `items` and `slidesToSlide` values for desktop to 3. Then you essentially have 3 "pages" each containing 3 slides. So it would be great to show 3 dots instead of 9, because you always slide a complete "page".

I guess to best this is to look at the react wrapper for the slick slider where this is shown.
https://react-slick.neostack.com/docs/example/multiple-items

**Comment by YIZHUANG** [comment]
@tpinne  Sounds good, i will find time to implement that, and PR is always welcome too!

**Comment by tpinne** [comment]
I would love to dive into this and make a PR. But sadly we are on very tight schedule with our project so we built a similar logic with a container component around your carousel and steering that.
Don't know if and when I would find the time for a PR :-(

**Comment by YIZHUANG** [comment]
@tpinne That means perfect sense cause the dots are basically "customButtonGroup" in this case https://github.com/YIZHUANG/react-multi-carousel/blob/master/stories/CustomArrows.js , you can use 
```
const ThreeDots = () => {
  this.CarouselRef.next(3);
 this.CarouselRef.previous(3);
}
```
And use local state to keep track which dot is active.

I think this is a good approach in your case for the time being.

**Comment by YIZHUANG** [comment]
The new version of dot mode is now published at version 2.0.  https://github.com/YIZHUANG/react-multi-carousel/commit/5e0b3646842394e0f9c9e1367c5640954218ca67

**Comment by tpinne** [comment]
I guess we can close this as the changes have been released.
